mono/metadata/class.c

```diff
@@ -8641,6 +8648,9 @@ gint32
 mono_array_element_size (MonoClass *ac)
 {
 	g_assert (ac->rank);
+	if (!ac->size_inited) {
```

Comment:
Must update doc string to mention locking changes.
This check must be G_UNLIKELY

Reply:
I'm concerned about a lot of potential extra work we might be doing in mono_class_setup_fields due to this.
Does it impact perf of a typeload heavy test like csc compiling corlib?